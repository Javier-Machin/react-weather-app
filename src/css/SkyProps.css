@-moz-keyframes spin {
  100% {
    -moz-transform: rotate(360deg);
  }
}
@-webkit-keyframes spin {
  100% {
    -webkit-transform: rotate(360deg);
  }
}
@keyframes spin {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@-moz-keyframes pulse {
  50% {
    -moz-transform: scale(0.96);
  }
  100% {
    -moz-transform: scale(1);
  }
}
@-webkit-keyframes pulse {
  50% {
    -webkit-transform: scale(0.96);
  }
  100% {
    -webkit-transform: scale(1);
  }
}
@keyframes pulse {
  50% {
    -webkit-transform: scale(0.96);
    transform: scale(0.96);
  }
  100% {
    -webkit-transform: scale(1);
    transform: scale(1);
  }
}

@keyframes scrolling {
  50% {
    background-position: -50px -150%;
  }
  100% {
    background-position: -100px -300%;
  }
}

.sun {
  position: absolute;
  top: -20%;
  right: -12%;
  transition: transform 1s, opacity 2s, width 2s, top 1s, right 1s;
  width: 30%;
  -webkit-animation: spin 25s linear infinite;
  -moz-animation: spin 25s linear infinite;
  animation: spin 25s linear infinite;
}

.sun-hidden {
  top: 14rem;
  right: -30rem;
  opacity: 0;
}

.sun-small {
  width: 25%;
}

.sun-big {
  width: 35%;
}

.cloud {
  position: absolute;
  top: 1%;
  transition: opacity 5s;
  width: 30%;
  -webkit-animation: pulse 4s linear infinite;
  -moz-animation: pulse 4s linear infinite;
  animation: pulse 4s linear infinite;
}

.cloud-dark {
  filter: brightness(50%);
}

.cloud-1 {
  transition: all 0.8s;
  left: 1%;
}

.cloud-2 {
  transition: all 2s;
  left: 21%;
}

.cloud-3 {
  transition: all 1.2s;
  left: 36%;
}

.cloud-4 {
  transition: all 2.5s;
  left: 56%;
}

.cloud-5 {
  transition: all 2.7s;
  left: 69%;
}

.hidden {
  left: -30rem !important;
  transition: all 2s;
  opacity: 0;
}

.rain {
  position: absolute;
  top: 22%;
  width: 110%;
  height: 53%;
  left: -5%;
  opacity: 0.6;
  transition: all 1.2s;
  background-size: 100%;
  background-repeat: repeat;
  animation: scrolling 4s linear infinite;
}

.rain-hidden {
  transform: translateY(-80rem);
  opacity: 0;
}

.display-filter {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 90.15%;
  background-color: black;
  opacity: 0;
  transition: all 1s;
  background-size: cover;
  border-radius: 25px 25px 15px 15px;
  z-index: 0;
}

@media only screen and (max-width: 650px) {
  .display-filter {
    border-radius: 0;
  }
}

.display-filter-dark {
  opacity: 0.19;
}

.display-filter-darker {
  opacity: 0.26;
}
