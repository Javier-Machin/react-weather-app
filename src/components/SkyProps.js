import React from 'react';
import Cloud from '../images/cloud-1.png';
import Sun from '../images/sun.png';
import Rain from '../images/rain.png';
import skyPropsHandler from '../utils/skyPropsHandler';
import '../css/SkyProps.css';

// Animated sky objects that change and move around depending on weather data

const SkyProps = props => {
  const { weatherData, statusCode } = props;
  // If the request was successful
  if (statusCode === 200) {
    // Get CSS classes for all props depending on weather
    const {
      cloudsClasses,
      sunClasses,
      displayFilterClasses,
      rainClasses
    } = skyPropsHandler(weatherData);

    // Set rain background image
    const rainStyle = { backgroundImage: `url(${Rain})` };

    return (
      <React.Fragment>
        <img className={sunClasses} src={Sun} alt="Sun drawing" />
        <div className={rainClasses} style={rainStyle} />
        <img className={cloudsClasses[0]} src={Cloud} alt="Small cloud" />
        <img className={cloudsClasses[1]} src={Cloud} alt="Small cloud" />
        <img className={cloudsClasses[2]} src={Cloud} alt="Small cloud" />
        <img className={cloudsClasses[3]} src={Cloud} alt="Small cloud" />
        <img className={cloudsClasses[4]} src={Cloud} alt="Small cloud" />
        <div className={displayFilterClasses} />
      </React.Fragment>
    );
  } else {
    return null;
  }
};

export default SkyProps;
