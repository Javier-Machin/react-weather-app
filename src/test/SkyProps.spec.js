import React from 'react';
import { shallow } from 'enzyme';
import SkyProps from '../components/SkyProps';

describe('SkyProps', () => {
  it('should render correctly', () => {
    const component = shallow(<SkyProps />);
    expect(component).toMatchSnapshot();
  });

  it('should display a big sun on sunny weather', () => {
    const weatherData = {
      temperature: 33,
      skyStatus: 'clear',
      rainProbability: 'unlikely'
    };

    const component = shallow(<SkyProps weatherData={weatherData} statusCode={200} />);
    expect(component.find('img.sun-big')).toHaveLength(1);
  });

  it('should hide rain on not rainy weather', () => {
    const weatherData = {
      temperature: 33,
      skyStatus: 'Overcast',
      rainProbability: 'unlikely'
    };

    const component = shallow(<SkyProps weatherData={weatherData} statusCode={200} />);
    expect(component.find('div.rain-hidden')).toHaveLength(1);
  });

  it('should not hide rain on rainy weather', () => {
    const weatherData = {
      temperature: 33,
      skyStatus: 'Overcast',
      rainProbability: 'probable'
    };

    const component = shallow(<SkyProps weatherData={weatherData} statusCode={200} />);
    expect(component.find('div.rain-hidden')).toHaveLength(0);
  });
});
