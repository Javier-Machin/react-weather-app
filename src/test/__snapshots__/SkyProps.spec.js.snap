// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyProps should render correctly 1`] = `""`;
